view/frontend/web/js/checkout-button.js

```diff
@@ -139,6 +143,69 @@ function ($, Component, placeAmwalOrder, payAmwalOrder, amwalErrorHandler, urlBu
                 attributes: false,
                 characterData: false
             });
+        },
+
+        /**
+         * initializeAmwalWidget
+         */
+
+        initializeAmwalWidget: function () {
```

Comment:
I see a lot of repeated code. Can the code be refactored and shared with the `initializeProductDetail` function?

Reply:
Yes, i will mere the `AmwalWidget` logic with `initializeProductDetail`